TRIDENT-15425 - Add "Degraded rules" section (#3003)
SumoLogic/sumologic-documentation: docs/cse/rules/before-writing-custom-rule.md

<img src={useBaseUrl('img/cse/example-in-editor.png')} alt="Example in editor" width="500"/>

## Degraded rules

> A degraded rule is one that has had a portion of the rule shut off because it causes the rule to exceed a processing limit. If you write a custom rule that becomes degraded, you must tune the rule to correct the problem.

Reviewer: ```suggestion
A degraded rule is one that has had a portion of the rule shut off to prevent it from exceeding a processing limit. If you write a custom rule that becomes degraded, you must tune the rule to correct the problem.
```

For example, rules have a limit on the number of records per second they can evaluate.  If there is a value used in the "group by" field that causes the rule to exceed that threshold, the particular value will be ignored, but the rest of the rule is still be used. In this case, Cloud SIEM might display a message like this:

> ```
> The aggregation on the group key 'admin@company.com' has a record volume exceeding the supported limit, and has been disabled. Consider tuning the rule to exclude records producing this group key.
> ```

Reviewer: I'd keep this in single backticks since the customer won't need to interact (copy/paste) the snippet. I'll bring up this topic at our mtg tomorrow to get your opinions!

```suggestion
`The aggregation on the group key 'admin@company.com' has a record volume exceeding the supported limit, and has been disabled. Consider tuning the rule to exclude records producing this group key.`
```

Author: That makes sense, @kimsauce.
